napari/components/_tests/test_layer_slicer.py

```diff
@@ -289,3 +292,30 @@ def test_slice_layers_async_with_one_3d_image(layer_slicer):
 
     layer_result = future.result()[layer]
     np.testing.assert_equal(layer_result.data, data[2, :, :])
+
+
+def test_slice_layers_async_with_one_3d_points(layer_slicer):
+    np.random.seed(0)
+    num_points = 100
+    data = np.rint(2.0 * np.random.rand(num_points, 3))
+    layer = Points(data=data)
+    lockable_internal_data = LockableData(layer._data)
+    layer._data = lockable_internal_data
+    dims = Dims(
+        ndim=3,
+        ndisplay=2,
+        range=((0, 3, 1), (0, 3, 1), (0, 3, 1)),
+        current_step=(1, 0, 0),
+    )
+
+    with lockable_internal_data.lock:
+        future = layer_slicer.slice_layers_async(layers=[layer], dims=dims)
+        assert not future.done()
+
+    layer_result = future.result()[layer]
+
+    _indices_view = np.array(layer_result.indices, dtype=int)
+    # get the selected points that are in view
+
+    # Target number of indices manually determined
+    np.testing.assert_equal(len(_indices_view), 48)
```

Comment:
Not sure I can stomach the 48 here. Even though it's reliable with the seed above, it hurts the readability of the test because you can't reason about why it should be 48 other than someone check that value at some point and blessed it. I think we would be better off manually defining a small set of points, and comparing the indices.

Reply:
That is fine. The locking fix makes it possible to do that.

Reply:
Hm, so would you suggest just moving https://github.com/napari/napari/blob/326ba76262166fbe3beb231a027726ef53350d8f/napari/layers/points/_tests/test_points.py#L2499 (which does test specific indices now) into this test?

Reply:
Works for me!

Reply:
Alternatively, we could *just* check the async functionality (e.g. everything up to `assert not future.done()` then throw away the rest) in this test module and keep the test of slicing logic in `test_points`?

You've mentioned test maintenance burden before so I'm curious what you think is actually the better option here: combine both tests into 1, or simplify `test_slice_layers_async_with_one_3d_points` to just test if the async slicing call returns immediately and test slicing logic in `test_points.py`.

Reply:
I'm also fine with only testing the async behavior here.